registry.go

```diff
@@ -74,82 +62,78 @@ func main() {
 	pool, err = pgx.NewConnPool(pgx.ConnPoolConfig{
 		ConnConfig:     pgxcfg,
 		MaxConnections: 20,
-		AfterConnect:   afterConnect,
+		AfterConnect: func(conn *pgx.Conn) error {
+			_, err := conn.Prepare("getPackage", `SELECT name, url FROM packages WHERE name = $1`)
+			return err
+		},
 	})
 	if err != nil {
 		log.Fatalf("Connection error: %s", err)
 	}
 	defer pool.Close()
 
-	port := os.Getenv("PORT")
-
-	if port == "" {
-		port = "3000"
+	binary, err := exec.LookPath("node")
+	if err != nil {
+		log.Fatalf("Could not lookup node path: %s", err)
 	}
 
-	binary, err := exec.LookPath("node")
-	checkError(err)
 	cmd := exec.Command(binary, "--expose_gc", "index.js")
 	env := os.Environ()
 	env = append([]string{"PORT=3001"}, env...)
 	cmd.Env = env
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	err = cmd.Start()
-	checkError(err)
+	if err := cmd.Start(); err != nil {
+		log.Fatalf("Could not start node: %s", err)
+	}
+	// TODO: Does defer even work here?
 	defer func() {
-		err = cmd.Wait()
-		checkError(err)
+		if err := cmd.Wait(); err != nil {
+			log.Fatalf("Node process failed: %s", err)
+		}
 	}()
 
-	proxy := goproxy.NewProxyHttpServer()
-
+	proxy = goproxy.NewProxyHttpServer()
 	proxy.Verbose = false
+	proxy.NonproxyHandler = http.HandlerFunc(NonProxy)
+	proxy.OnRequest(PathIs("/packages")).DoFunc(ListPackages)
+	proxy.OnRequest(UrlHasPrefix("/packages/")).DoFunc(GetPackage)
 
-	proxy.NonproxyHandler = http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		req.Host = "bower.herokuapp.com"
-		req.URL.Scheme = "http"
-		req.URL.Host = "localhost:3001"
-		proxy.ServeHTTP(w, req)
-	})
+	port := getEnv("PORT", "3000")
+	log.Println("Starting web server at port", port)
+	log.Fatal(http.ListenAndServe(":"+port, proxy))
```

Comment:
The final statement here is a `log.Fatal` which calls `os.Exit()`.  The `defer` functions will not run.